From ca85908ca6136f074277c078d66b34d1cfac9a66 Mon Sep 17 00:00:00 2001
From: Rui Wang <ruwang@redhat.com>
Date: Thu, 21 Aug 2014 04:32:11 -0400
Subject: [x86] avx-512: AVX-512 Feature Detection

Message-id: <1408595532-28320-2-git-send-email-ruwang@redhat.com>
Patchwork-id: 88502
O-Subject: [PATCH RHEL7.1 BZ1085563 1/2] x86, AVX-512: AVX-512 Feature Detection
Bugzilla: 1085563
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

From: "Wang, Xiaolong" <xiaolong.wang@intel.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1085563

Backport of:
Original commit 8e5780fdeef7dc490b3f0b3a62704593721fa4f3
Original Author: Fenghua Yu <fenghua.yu@intel.com>
Original Date:   Thu Feb 20 13:24:50 2014 -0800

    x86, AVX-512: AVX-512 Feature Detection

    AVX-512 is an extention of AVX2. Its spec can be found at:
    http://download-software.intel.com/sites/default/files/managed/71/2e/319433-017.pdf

    This patch detects AVX-512 features by CPUID.

Signed-off-by: Wang, Xiaolong <xiaolong.wang@intel.com>
Signed-off-by: Wang, Rui <ruwang@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/cpufeature.h b/arch/x86/include/asm/cpufeature.h
index 556aaff..b337e66 100644
--- a/arch/x86/include/asm/cpufeature.h
+++ b/arch/x86/include/asm/cpufeature.h
@@ -216,10 +216,14 @@
 #define X86_FEATURE_ERMS (9*32+ 9) /* Enhanced REP MOVSB/STOSB */
 #define X86_FEATURE_INVPCID (9*32+10) /* Invalidate Processor Context ID */
 #define X86_FEATURE_RTM  (9*32+11) /* Restricted Transactional Memory */
+#define X86_FEATURE_AVX512F (9*32+16) /* AVX-512 Foundation */
 #define X86_FEATURE_MPX  (9*32+14) /* Memory Protection Extension */
 #define X86_FEATURE_RDSEED (9*32+18) /* The RDSEED instruction */
 #define X86_FEATURE_ADX  (9*32+19) /* The ADCX and ADOX instructions */
 #define X86_FEATURE_SMAP (9*32+20) /* Supervisor Mode Access Prevention */
+#define X86_FEATURE_AVX512PF (9*32+26) /* AVX-512 Prefetch */
+#define X86_FEATURE_AVX512ER (9*32+27) /* AVX-512 Exponential and Reciprocal */
+#define X86_FEATURE_AVX512CD (9*32+28) /* AVX-512 Conflict Detection */
 
 /*
  * BUG word(s)
-- 
1.7.1